exonum-java-binding/core/rust/ejb-app/src/node_builder.rs

```diff
@@ -76,81 +63,81 @@ pub fn create() -> fabric::NodeBuilder {
+        assert!(services.contains(CONFIGURATION_SERVICE));
     }
 
     #[test]
-    fn duplicated() {
-        let cfg = create_config(
-            "duplicated.toml",
-            "services = [\"btc-anchoring\", \"btc-anchoring\"]",
-        );
-        let services_to_enable = services_to_enable(cfg);
-        assert_eq!(services_to_enable.len(), 2);
-        assert!(services_to_enable.contains(EJB_SERVICE));
-        assert!(services_to_enable.contains(BTC_ANCHORING_SERVICE));
-    }
-
-    #[test]
-    #[should_panic(expected = "Invalid list of services to enable")]
-    fn broken_config() {
-        let cfg = create_config("broken.toml", "not_list = 1");
-        let _services_to_enable = services_to_enable(cfg);
-    }
-
-    #[test]
-    fn with_anchoring() {
-        let cfg = create_config("anchoring.toml", "services = [\"btc-anchoring\"]");
-        let services_to_enable = services_to_enable(cfg);
-        assert_eq!(services_to_enable.len(), 2);
-        assert!(services_to_enable.contains(EJB_SERVICE));
-        assert!(services_to_enable.contains(BTC_ANCHORING_SERVICE));
+    fn validate_services_ok() {
+        let mut services = HashSet::new();
+        services.insert(BTC_ANCHORING_SERVICE.to_owned());
+        services.insert(TIME_SERVICE.to_owned());
+        let services = validate_services(Some(services));
+        assert_eq!(services.len(), 2);
+        assert!(services.contains(BTC_ANCHORING_SERVICE));
+        assert!(services.contains(TIME_SERVICE));
     }
 
     #[test]
     fn all_services() {
         let cfg = create_config(
             "all.toml",
-            "services = [\"configuration\", \"btc-anchoring\", \"time\"]",
+            "enabled_services = [\"configuration\", \"btc-anchoring\", \"time\"]\n\
+            [user_services]\nservice_name1 = '/path/to/artifact1'\nservice_name2 = '/path/to/artifact2'",
         );
-        let services_to_enable = services_to_enable(cfg);
-        assert_eq!(services_to_enable.len(), 4);
-        assert!(services_to_enable.contains(EJB_SERVICE));
-        assert!(services_to_enable.contains(CONFIGURATION_SERVICE));
-        assert!(services_to_enable.contains(BTC_ANCHORING_SERVICE));
-        assert!(services_to_enable.contains(TIME_SERVICE));
+        let EjbAppServices {
+            enabled_services,
+            user_services,
+        } = load_services_definition(cfg).unwrap();
+        let enabled_services = enabled_services.unwrap();
+        assert_eq!(enabled_services.len(), 3);
+        assert!(enabled_services.contains(CONFIGURATION_SERVICE));
+        assert!(enabled_services.contains(BTC_ANCHORING_SERVICE));
+        assert!(enabled_services.contains(TIME_SERVICE));
 
         let service_factories = service_factories();
-        for service in &services_to_enable {
-            assert!(service_factories.get(service).is_some())
+        for service in enabled_services {
+            assert!(service_factories.get(&service).is_some())
```

Comment:
Not sure about the original meaning of this part because I just adapted the old test in node_builder but I agree that it looks redundant within this test.

Reply:
The majority of `node_builder` code is rewritten in order to make it more testable.